ui-particles-angular/projects/ui-particles-angular/src/lib/oem-theme/gio-menu/gio-menu-search/gio-menu-search.component.ts

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright (C) 2015 The Gravitee team (http://gravitee.io)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { Component, EventEmitter, OnInit, Output } from '@angular/core';
+import { EMPTY, Observable } from 'rxjs';
+import { FormControl } from '@angular/forms';
+import { map, startWith } from 'rxjs/operators';
+import { MatAutocompleteSelectedEvent } from '@angular/material/autocomplete';
+
+import { GioMenuSearchService } from './gio-menu-search.service';
+
+export interface MenuSearchItem {
+  name: string;
+  routerLink: string;
+  category: string;
+  groupIds?: string[];
+}
+
+@Component({
+  selector: 'gio-menu-search',
+  templateUrl: './gio-menu-search.component.html',
+  styleUrls: ['./gio-menu-search.component.scss'],
+})
+export class GioMenuSearchComponent implements OnInit {
+  protected control = new FormControl<string | MenuSearchItem>('');
+  protected filteredItems$: Observable<MenuSearchItem[]> = EMPTY;
+
+  @Output()
+  public valueChanges: EventEmitter<MenuSearchItem> = new EventEmitter<MenuSearchItem>();
+
+  constructor(private readonly gioMenuSearchService: GioMenuSearchService) {}
+
+  public ngOnInit() {
+    this.filteredItems$ = this.control.valueChanges.pipe(
+      startWith(''),
+      map(value => {
+        const name = typeof value === 'string' ? value : value?.name;
+        return this._filter(name);
+      }),
+    );
+  }
+
+  protected displayFn(item: MenuSearchItem): string {
+    return item && item.name ? item.name : '';
```

Comment:
This should also work. But I'll let you decide if it worth the change 🙈 

```suggestion
    return item?.name ?? ''
```